src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -285,63 +285,117 @@ def update_app_service_plan(instance, sku=None, number_of_workers=None,
-    location = _get_location_from_webapp(client, resource_group_name, name)
-    db_setting = DatabaseBackupSetting(database_type, database_name,
-                                       connection_string=database_connection_string)
+    if backup_name is not None and backup_name.lower().endswith('.zip'):
+        backup_name = backup_name[:-4]
+    location = _get_location_from_webapp(client, resource_group_name, webapp_name)
+    db_setting = _create_db_setting(db_name, db_type, db_connection_string)
     backup_request = BackupRequest(location, backup_request_name=backup_name,
-                                   storage_account_url=storage_account_url, databases=[db_setting])
-    if slot is None:
-        return client.sites.backup_site(resource_group_name, name, backup_request)
+                                   storage_account_url=storage_account_url, databases=db_setting)
+    if slot:
+        return client.sites.backup_site_slot(resource_group_name, webapp_name, backup_request, slot)
     else:
-        return client.sites.backup_site_slot(resource_group_name, name, backup_request, slot)
+        return client.sites.backup_site(resource_group_name, webapp_name, backup_request)
 
-def update_backup_schedule(resource_group_name, name, storage_account_url,
-                           schedule_frequency, schedule_frequency_unit,
-                           keep_at_least_one_backup=None,
-                           retention_period_in_days=90, database_name=None,
-                           database_connection_string=None, database_type=None, slot=None):
+def update_backup_schedule(resource_group_name, webapp_name, storage_account_url=None,
+                           frequency=None, keep_at_least_one_backup=False,
+                           retention_period_in_days=None, db_name=None,
+                           db_connection_string=None, db_type=None, slot=None):
     client = web_client_factory()
-    location = _get_location_from_webapp(client, resource_group_name, name)
-    db_setting = DatabaseBackupSetting(database_type, database_name,
-                                       connection_string=database_connection_string)
-    backup_schedule = BackupSchedule(schedule_frequency_unit, schedule_frequency,
+    location = _get_location_from_webapp(client, resource_group_name, webapp_name)
+    configuration = None
+
+    try:
+        configuration = _generic_site_operation(resource_group_name, webapp_name, 'get_site_backup_configuration', slot)
+    except:
+        # No configuration set yet
+        if not storage_account_url or not frequency or not retention_period_in_days:
+            raise CLIError('No backup configuration found. Specify container-url, frequency, and retention period to create one.')
+
+    storage_account_url = storage_account_url or configuration.storage_account_url
+    if keep_at_least_one_backup is None:
+        keep_at_least_one_backup = configuration.backup_schedule.keep_at_least_one_backup
```

Comment:
I think there's an error here, but it's actually that we want to always default to false unless the user specifies --retain-one. It doesn't make much sense to fall back to what's already in the backup configuration because --retain-one is a true/false flag